index.js

```diff
@@ -29,16 +29,19 @@ var SPLIT_REGEX = /[^.^\]^[]+|(?=\[\]|\.\.)/g,
   CLEAN_QUOTES_REGEX = /^\s*(['"]?)(.*?)(\1)\s*$/,
   MAX_CACHE_SIZE = 512
 
-var contentSecurityPolicy = false,
+var contentSecurityPolicyEnabled = true,
   pathCache = new Cache(MAX_CACHE_SIZE),
   setCache = new Cache(MAX_CACHE_SIZE),
   getCache = new Cache(MAX_CACHE_SIZE)
 
-try {
-  new Function('')
-} catch (error) {
-  contentSecurityPolicy = true
-}
+if (process.env.NODE_ENV !== 'production' || process.env.CSP_DISABLED) {
```

Comment:
we shouldn't use a custom "environment" variable for this as this would require custom build setups to enable, I think if anything this should be a runtime configuration, at least in the short term. I'm not convinced the CSP version is the better default, it's a good bit slower. and if we were using the CSP version for production there isn't really any value in maintaining the other version

Reply:
Hmm, the only other way I see to remove the errors would be to export getter and setter in separate (CSP safe and non safe) versions - all modern compilers will automatically remove unused version in builds anyway, and that will solve the issue without relaying on env variables. Will update the PR in a bit :)

Regarding whether which version should be a default one  - I guess with rules getting stricter over time problem will be affecting more and more users, and safer version should be the default one, but obviously the choice is yours :)

Reply:
My point is generally that it doesn't make sense to change library core functionality based on build target. you, ideally, want the same functional code in dev and prod, lest you run into bugs in the prod branch you can't test in dev. 

In this case this is a concern purely for folks using this CSP rule, the code run here isn't actually unsafe in practice. I'm happy to move to a purely CSP friendly version providing we can demonstrate the perf profile is reasonable for the none "compiled" version

Reply:
I agree, however, did you check update of the pr? 
It will no longer produce different output depending on env - lib would now just give the users an option whether to use CSP-safe solution or not.

Will try running some profiling on both solutions and see how they perform against each other.

Reply:
A CSP-safe solution by default with an opt-out to the more performant solution seems like the way to go. For me this module is a dependency of a dependency so I can't easily choose which implementation will be used by my module. 

Currently, my CSP reporting tool is lighting up continuously due to this issue so I'm highly anticipating an outcome.

Reply:
the CSP-safe version is already the default in CSP environments. I'm not actually sure i understand why this is required, we use expr extensively along with CSP and have not run into any issues

Reply:
A CSP report is still generated even though the error is caught and the code works. Since this is a "known" issue that affects every page, it tends to drown out actual CSP violations.

In addition, Chrome seems to be the only browser that doesn't emit a console error/warning: you can still see the error emitted in Firefox and Edge.

Reply:
As the previous poster said, the problem here is not that the CSP version is not working, but that each CSP violation generates a report to your configured `report-uri`,  and as the check for CSP is done in the main body, it is executed each time the JS file is loaded by a browser and generates a CSP report violation.
Most CSP reporting tools can filter CSP reports based on content, but as this is reported as an inline code execution, location is always line 1 / character 1, so you can not differentiate between this "legit" violation or another violation, making the reporting quite useless.

I would love to have a way to say that I am using a strict CSP policy and I never want this executed, opting for the slower path.

Reply:
Just chiming in to confirm that this issue still persists, and if this package is in a bundle served with a strict CSP + `report-uri` configured, a report gets filed from the browser literally every time a user visits the page. I'm currently looking at hundreds of thousands of violations in my aggregation dashboard, solely due to this package.

Reply:
pls guys, merge it